keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/SimpleDoubleTensor.java

```diff
@@ -0,0 +1,319 @@
+package io.improbable.keanu.vertices.dbltensor;
+
+public class SimpleDoubleTensor implements DoubleTensor {
+
+    private Double value;
+    private int[] shape;
+
+    public SimpleDoubleTensor(double value) {
+        this.value = value;
+        this.shape = SCALAR_SHAPE;
+    }
+
+    public SimpleDoubleTensor(int[] shape) {
+        this.value = null;
+        this.shape = shape;
+    }
+
+    private SimpleDoubleTensor duplicate() {
+        return new SimpleDoubleTensor(value);
+    }
+
+    @Override
+    public double getValue(int[] index) {
+        return value;
+    }
+
+    @Override
+    public void setValue(double value, int[] index) {
+        this.value = value;
+    }
+
+    @Override
+    public double sum() {
+        return value;
+    }
+
+    @Override
+    public double scalar() {
+        return value;
+    }
+
+    @Override
+    public DoubleTensor reciprocal() {
+        return this.duplicate().reciprocalInPlace();
```

Comment:
Why have we changed this to `duplicate` and then `op`? Is it faster?

If so, should we apply the `duplicate` then `op` logic to  `Nd4jDoubleTensor` operations to?

Reply:
It's not necessarily faster. It's probably about the same, just less code. Good idea about doing the same in `Nd4jDoubleTensor`. I've updated the applicable methods there.